dll/win32/kernel32/client/session.c

```diff
@@ -13,31 +16,124 @@
 #define NDEBUG
 #include <debug.h>
 
+/* GLOBALS ********************************************************************/
+
+PBASE_STATIC_SERVER_DATA BaseStaticServerData;
+
+#define LMEM_FIXED 0x0
+
+#define ERROR_INVALID_PARAMETER 0x57
+#define ERROR_NOT_ENOUGH_MEMORY 0x8
+
 /* FUNCTIONS ******************************************************************/
 
 /*
- * @unimplemented
+ * @implemented
  */
 DWORD
 WINAPI
 DosPathToSessionPathW(IN DWORD SessionID,
                       IN LPWSTR InPath,
                       OUT LPWSTR *OutPath)
 {
-    UNIMPLEMENTED;
+    DWORD LastError = 0;
+    if ((InPath == 0x0) || IsBadReadPtr(InPath, 2), IsBadWritePtr(OutPath, 8)) {
```

Comment:
IsBad*Ptr functions are deprecated and shouldn't be used.
https://devblogs.microsoft.com/oldnewthing/20060927-07/?p=29563